frida_tools/tracer.py

```diff
@@ -68,14 +69,21 @@ def _initialize(self, parser, options, args):
             self._decorate = options.decorate
             self._output = None
             self._output_path = options.output
+            self._userdata = self.escape_char(options.userdata, '"')
```

Comment:
How about parsing this as JSON instead?

Reply:
Great minds think alike!  Here is newer code I have been using (modified per using `rpc.exports`:

```
        return """'use strict';

var started = Date.now();
var handlers = {};
var state = {};
var pending = [];
var timer = null;

state.newline = String.fromCharCode(10);

try {
    state.userdata_json_object = JSON.parse(state.userdata);
} catch (e) {
    // user data is not valid JSON, leave it as-is
}
```
We don't want to force the user to pass JSON data via the `-userdata` option, but we should make his life easier if he does.  So if a user knows he's passing JSON via `userdata`, he can expect to find a valid variable `state.userdata_json_object`, if the data is valid.  Of course, `state.userdata` exists in any case.

What are your thoughts?

Reply:
Please see the next issue, I'm awaiting your decision.